### PR TITLE
Fix another lifecycle issue

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This covers the case where you manually change the URL when DevTools is run on flutter web. Whenever we navigate to the landing screen, we should disconnect from any connected app.

### CODE REVIEW

File: packages/devtools_app/lib/src/app.dart [hunk 1]
```diff
@@ -162,6 +162,8 @@ class DevToolsAppState extends State<DevToolsApp> {
 
     // Always return the landing screen if there's no VM service URI.
     if (vmServiceUri?.isEmpty ?? true) {
+      // Ensure we disconnect from any previously connected apps.
+      serviceManager.manuallyDisconnect();
```
Comment (jacob314) [inline packages/devtools_app/lib/src/app.dart:166]: building a widget shouldn't have side effects like this. Is there a different fix?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart]: Moved disconnect calls into router. Fixing tests, then will land so I can get this into the dev release.

File: packages/devtools_app/lib/src/scaffold.dart [hunk 2]
```diff
@@ -327,6 +327,10 @@ class DevToolsScaffoldState extends State<DevToolsScaffold>
                   children: tabBodies,
                 ),
                 if (serviceManager.hasConnection &&
+                    DevToolsRouterDelegate.of(context)
```
Comment (jacob314) [inline packages/devtools_app/lib/src/scaffold.dart:330]: Add a TODO to clean this up. Seems odd we have to bug the router for this.

**Review by jacob314 — APPROVED** [review]
lgtm

**Comment by kenzieschmoll** [comment]
Landing on red. Test failure is from memory screenshot tests failing with 0.00% diff...

**Pull request merged**